### PR TITLE
Salesforce plugins does not support Array types

### PR DESCRIPTION
**Pull request by sourabhrawat345**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
     .put(FieldType.datetime, Schema.of(Schema.LogicalType.TIMESTAMP_MICROS))
     .put(FieldType.time, Schema.of(Schema.LogicalType.TIME_MICROS))
     .put(FieldType.string, Schema.of(Schema.Type.STRING))
+    .put(FieldType.complexvalue, Schema.arrayOf(Schema.of(Schema.Type.STRING)))
```
Comment (snehalakshmisha) [inline src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java:56]: Does this also support array of types other than string?

**Review by snehalakshmisha — COMMENTED** [review]
Can you add testing details?

File: src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java (hunk 1 above)
Comment (sourabhrawat345) [reply src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java:56]: yes this is supporting other than string.

Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java:56]: Yes, it is also supporting arrays or other types, that implementation is written in getSchemaWithFields method.

**Comment by sourabhrawat345** [comment]
> Can you add testing details?

what all the details you need for this ?

**Comment by poojantcs** [comment]
> Can you add testing details?

Hi Sneha, 

We executed a few relationship queries to get 'array' type in the Output schema table. 
Eg. SELECT Name, Phone, BillingCountry, (SELECT Id, Name, Phone FROM Contacts), (SELECT Id, Name FROM Opportunities) FROM Account

We verified the Salesforce batch source (Output Schema table), BigQuery sink (Input Schema table) and BigQuery Schema and table for the array type data.

**Comment by snehalakshmisha** [comment]
> > Can you add testing details?
> 
> Hi Sneha,
> 
> We executed a few relationship queries to get 'array' type in the Output schema table. Eg. SELECT Name, Phone, BillingCountry, (SELECT Id, Name, Phone FROM Contacts), (SELECT Id, Name FROM Opportunities) FROM Account
> 
> We verified the Salesforce batch source (Output Schema table), BigQuery sink (Input Schema table) and BigQuery Schema and table for the array type data.

Can you please add a screenshot for the same with at least one row in the table?

**Comment by poojantcs** [comment]
> > > Can you add testing details?
> > 
> > 
> > Hi Sneha,
> > We executed a few relationship queries to get 'array' type in the Output schema table. Eg. SELECT Name, Phone, BillingCountry, (SELECT Id, Name, Phone FROM Contacts), (SELECT Id, Name FROM Opportunities) FROM Account
> > We verified the Salesforce batch source (Output Schema table), BigQuery sink (Input Schema table) and BigQuery Schema and table for the array type data.
> 
> Can you please add a screenshot for the same with at least one row in the table?

Hi @snehalakshmisha , 
Please find the screenshots for each of the verification points:

Source-Output Schema 
Sink(BQ)-Input Schema
BQ-Schema
BQ-Table

**Review by snehalakshmisha — APPROVED** [review]
lgtm

**Review by rmstar — COMMENTED** [review]
Is this fix applicable to salesforce batch source and sink? If yes, please update the documentation for those plugins

**Comment by sourabhrawat345** [comment]
> Is this fix applicable to salesforce batch source and sink? If yes, please update the documentation for those plugins

as of now it's only supporting salesforce batch source and the document has been updated for this.

**Pull request merged**